Xplat/src/main/java/vazkii/botania/common/block/mana/DrumBlock.java

```diff
@@ -64,26 +70,90 @@ public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, Co
+				GATHER_RANGE + 1, GATHER_RANGE + 1, GATHER_RANGE + 1)), e -> !BergamuteBlockEntity.isBergamuteNearby(world, e.getX(), e.getY(), e.getZ()));
+		List<Shearable> shearables = new ArrayList<>();
+
+		for (Mob entity : entities) {
+			if (entity.getType().is(BotaniaTags.Entities.DRUM_MILKABLE) && !entity.isBaby()) {
+				convertNearby(entity, Items.BUCKET, Items.MILK_BUCKET);
+			}
+			if (entity instanceof MushroomCow mooshroom && !mooshroom.isBaby()) {
+				if (mooshroom.getVariant() == MushroomCow.MushroomType.BROWN && ((MushroomCowAccessor) mooshroom).getEffect() != null) {
+					fillBowlSuspiciously(mooshroom);
+				}
+				convertNearby(entity, Items.BOWL, Items.MUSHROOM_STEW);
+			}
+			if (entity instanceof Shearable shearable && !entity.getType().is(BotaniaTags.Entities.DRUM_NO_SHEARING) && shearable.readyForShearing()) {
+				shearables.add(shearable);
+			}
+		}
+
+		Collections.shuffle(shearables);
+		int sheared = 0;
+
+		for (Shearable shearable : shearables) {
+			if (sheared > MAX_NUM_SHEARED) {
+				break;
+			}
+
+			shearable.shear(SoundSource.BLOCKS);
+			++sheared;
+		}
+	}
+
 	private static void convertNearby(Entity entity, Item from, Item to) {
 		Level world = entity.level();
 		List<ItemEntity> items = world.getEntitiesOfClass(ItemEntity.class, entity.getBoundingBox());
 		for (ItemEntity item : items) {
 			ItemStack itemstack = item.getItem();
-			if (!itemstack.isEmpty() && itemstack.is(from) && !world.isClientSide) {
+			if (!itemstack.isEmpty() && itemstack.is(from)) {
 				while (itemstack.getCount() > 0) {
-					ItemEntity ent = entity.spawnAtLocation(new ItemStack(to), 1.0F);
-					ent.setDeltaMovement(ent.getDeltaMovement().add(
-							world.random.nextFloat() * 0.05F,
-							(world.random.nextFloat() - world.random.nextFloat()) * 0.1F,
-							(world.random.nextFloat() - world.random.nextFloat()) * 0.1F
-					));
+					spawnItem(entity, new ItemStack(to));
 					itemstack.shrink(1);
 				}
 				item.discard();
 			}
 		}
 	}
 
+	private static void spawnItem(Entity entity, ItemStack to) {
+		Level world = entity.level();
+		ItemEntity ent = entity.spawnAtLocation(to, 1.0F);
+		ent.setDeltaMovement(ent.getDeltaMovement().add(
+				world.random.nextFloat() * 0.05F,
+				(world.random.nextFloat() - world.random.nextFloat()) * 0.1F,
+				(world.random.nextFloat() - world.random.nextFloat()) * 0.1F
+		));
+	}
+
+	private static void fillBowlSuspiciously(MushroomCow mushroomCow) {
+		MushroomCowAccessor mushroomCowAccessor = (MushroomCowAccessor) mushroomCow;
+		MobEffect effect = mushroomCowAccessor.getEffect();
+		int effectDuration = mushroomCowAccessor.getEffectDuration();
+
+		Level world = mushroomCow.level();
+		List<ItemEntity> items = world.getEntitiesOfClass(ItemEntity.class, mushroomCow.getBoundingBox(),
+				itemEntity -> itemEntity.getItem().is(Items.BOWL) && !itemEntity.getItem().isEmpty());
+		for (ItemEntity item : items) {
+			ItemStack itemstack = item.getItem();
```

Comment:
let's call this something with `bowl` in it so it's clear which is which, it's easy to get mixed up when everything is `item` or `stack` or some combo